xilem_core/tests/one_of.rs

```diff
@@ -0,0 +1,210 @@
+    let mut ctx = TestCtx::default();
+    let (mut element, mut state) = view1.build(&mut ctx);
+    ctx.assert_empty();
+    assert_eq!(element.operations, &[Operation::Build(0)]);
+
+    view1.teardown(&mut state, &mut ctx, &mut element);
+    ctx.assert_empty();
+    assert_eq!(element.view_path[0], ViewId::new(0));
+    assert_eq!(
+        element.operations,
+        &[Operation::Build(0), Operation::Teardown(0)]
+    );
+}
+
+#[test]
+fn one_of_passthrough_message() {
+    let view1: OneOf2<OperationView<0>, OperationView<1>> = OneOf2::A(record_ops_0(0));
+    let mut ctx = TestCtx::default();
+    let (element, mut state) = view1.build(&mut ctx);
+    ctx.assert_empty();
+    assert_eq!(element.operations, &[Operation::Build(0)]);
+
+    let result = view1.message(&mut state, &element.view_path, Box::new(()), &mut ());
+    assert_action(result, 0);
+}
+
+#[test]
+fn one_of_no_message_after_stale() {
+    let view1 = OneOf2::A(OperationView::<0>(0));
+    let mut ctx = TestCtx::default();
+    let (mut element, mut state) = view1.build(&mut ctx);
+    ctx.assert_empty();
+    let path = element.view_path.clone();
+
+    let view2 = OneOf2::B(OperationView::<1>(1));
+    view2.rebuild(&view1, &mut state, &mut ctx, &mut element);
+    ctx.assert_empty();
+    assert_eq!(
+        element.operations,
+        &[
+            Operation::Build(0),
+            Operation::Teardown(0),
+            Operation::Replace(1)
+        ]
+    );
+
+    let result = view2.message(&mut state, &path, Box::new(()), &mut ());
+    assert!(matches!(result, MessageResult::Stale(_)));
+}
+
+#[test]
+fn one_of_no_message_after_stale_then_same_type() {
+    let view1 = OneOf2::A(OperationView::<0>(0));
+    let mut ctx = TestCtx::default();
+    let (mut element, mut state) = view1.build(&mut ctx);
+    ctx.assert_empty();
+    let path = element.view_path.clone();
+
+    let view2 = OneOf2::B(OperationView::<1>(1));
+    view2.rebuild(&view1, &mut state, &mut ctx, &mut element);
+    ctx.assert_empty();
+    assert_eq!(
+        element.operations,
+        &[
+            Operation::Build(0),
+            Operation::Teardown(0),
+            Operation::Replace(1)
+        ]
+    );
+
+    let view3 = OneOf2::A(OperationView::<0>(2));
+    view3.rebuild(&view2, &mut state, &mut ctx, &mut element);
+    ctx.assert_empty();
+    assert_eq!(
+        element.operations,
+        &[
+            Operation::Build(0),
+            Operation::Teardown(0),
+            Operation::Replace(1),
+            Operation::Teardown(1),
+            Operation::Replace(2)
+        ]
+    );
+
+    let result = view3.message(&mut state, &path, Box::new(()), &mut ());
+    assert!(matches!(result, MessageResult::Stale(_)));
+}
+
+// TODO: Logic for the `ViewSequence` implementation of `OneOf` is basically the same as the view minus up/downcasting of the element type
+// Tests would be great, but probably not necessary
```

Comment:
Well this didn't age well... As I just found an issue while manually testing this. I mean now it should probably be correct, but yeah tests would be great. (I just hate writing tests...)